src/main/scala/it/bitrock/dvs/streams/topologies/FlightInterpolatedListStream.scala

```diff
@@ -0,0 +1,104 @@
+package it.bitrock.dvs.streams.topologies
+
+import java.util.Properties
+
+import it.bitrock.dvs.model.avro.{FlightReceived, FlightReceivedList}
+import it.bitrock.dvs.streams.StreamProps.streamProperties
+import it.bitrock.dvs.streams._
+import it.bitrock.dvs.streams.config.AppConfig
+import it.bitrock.dvs.streams.geo.utils.EarthPositionCalculator
+import org.apache.kafka.common.serialization.Serde
+import org.apache.kafka.streams.kstream.Transformer
+import org.apache.kafka.streams.processor._
+import org.apache.kafka.streams.scala.ImplicitConversions._
+import org.apache.kafka.streams.scala.StreamsBuilder
+import org.apache.kafka.streams.state.{KeyValueStore, Stores}
+import org.apache.kafka.streams.{KeyValue, Topology}
+
+import scala.concurrent.duration._
+
+object FlightInterpolatedListStream {
+  private val currentSnapshot = "currentSnapshot"
+
+  def buildTopology(config: AppConfig, kafkaStreamsOptions: KafkaStreamsOptions): List[(Topology, Properties)] = {
+    implicit val stringKeySerde: Serde[String]                           = kafkaStreamsOptions.stringKeySerde
+    implicit val flightReceivedListEventSerde: Serde[FlightReceivedList] = kafkaStreamsOptions.flightReceivedListEventSerde
+
+    val streamsBuilder = new StreamsBuilder
+
+    val flightListInterpolationStateStoreName = "FlightListInterpolationStateStore"
+    val stateStore = Stores.keyValueStoreBuilder(
+      Stores.persistentKeyValueStore(flightListInterpolationStateStoreName),
+      stringKeySerde,
+      flightReceivedListEventSerde
+    )
+
+    streamsBuilder.addStateStore(stateStore)
+
+    streamsBuilder
+      .stream[String, FlightReceivedList](config.kafka.topology.flightEnRouteListTopic.name)
+      .transform[String, FlightReceivedList](
+        () => interpolationTransformer(stateStore.name, config.kafka.topology.interpolationInterval),
+        stateStore.name
+      )
+      .to(config.kafka.topology.flightInterpolatedListTopic.name)
+
+    val props = streamProperties(config.kafka, config.kafka.topology.flightReceivedListTopic.name)
+    List((streamsBuilder.build(props), props))
+  }
+
+  @SuppressWarnings(Array("DisableSyntax.var"))
+  private def interpolationTransformer(
+      stateStoreName: String,
+      interpolationInterval: FiniteDuration
+  ): Transformer[String, FlightReceivedList, KeyValue[String, FlightReceivedList]] =
+    new Transformer[String, FlightReceivedList, KeyValue[String, FlightReceivedList]] {
+      private var processorContext: ProcessorContext                       = _
```

Comment:
The field isn't used. Can we throw it away or it enforces side effects?

Reply:
Good catch, an oversight of a previous implementation